### DISCUSSION TITLE
Retrieve phone-number of incoming call

### DISCUSSION THREAD

**Discussion by ddxe** [pnp/generator-teams#209] [Q&A] [answered]
Hi!

Can someone point me in the right direction on how to retrieve the phone number when someone calls a teams-connected phone number? 

I'm looking into the feasibility of creating an app/bot which triggers when someone external calls to a number connected to teams and we would then call an external api to a CRM and show a card/tab with information and options (lookup customer/create ticket)

**Comment by appieschot** [comment, marked as answer]
Basic details are available on https://docs.microsoft.com/en-us/microsoftteams/platform/bots/calls-and-meetings/calls-meetings-bots-overview including some samples. Setting up the bot docs (and pre-reqs) are available on: https://docs.microsoft.com/en-us/graph/cloud-communications-phone-number if you are looking for background info. 

The Graph then allows you to use the `/communications/calls/{id}/participants` to list all participants: https://docs.microsoft.com/en-us/graph/api/call-list-participants?view=graph-rest-1.0&tabs=http
